### PR TITLE
added esConsumes to modules in RecoHGCal/TICL and related packages

### PR DESCRIPTION
**Pull request by JamminJones**
#### PR description:

while adding esConsumes to the modules the RecHitTools class was modified to decouple EventSetup from it.

#### PR validation:

the code compiles

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by Dr15Jones** [comment]
This work was done under my supervision.

**Comment by civanch** [comment]
+1

**Comment by rovere** [comment]
> This work was done under my supervision.

Thanks to you two both!

File: RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h [hunk 1]
```diff
@@ -23,8 +25,9 @@ namespace hgcal {
     RecHitTools() : geom_(nullptr), fhOffset_(0), bhOffset_(0), fhLastLayer_(0), noseLastLayer_(0), geometryType_(0) {}
     ~RecHitTools() {}
 
-    void getEvent(const edm::Event&);
-    void getEventSetup(const edm::EventSetup&);
+    //void getEvent(const edm::Event&);
```
Comment (kpedro88) [inline RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h:28]: delete commented-out code

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 2]
```diff
@@ -65,13 +65,13 @@ namespace {
 
 }  // namespace
 
-void RecHitTools::getEvent(const edm::Event& ev) {}
+//void RecHitTools::getEvent(const edm::Event& ev) {}
```
Comment (kpedro88) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc:68]: delete commented-out code

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 3]
```diff
@@ -65,13 +65,13 @@ namespace {
...
 
-void RecHitTools::getEventSetup(const edm::EventSetup& es) {
-  edm::ESHandle<CaloGeometry> geom;
-  es.get<CaloGeometryRecord>().get(geom);
+void RecHitTools::setGeometry(const CaloGeometry& geom) {
+  //edm::ESHandle<CaloGeometry> geom;
```
Comment (kpedro88) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc:71]: delete commented-out code

**Comment by Dr15Jones** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by Dr15Jones** [comment]
@cms-sw/simulation-l2 ping

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**